src/extensions/extension-discovery.ts

```diff
@@ -366,11 +366,8 @@ export class ExtensionDiscovery extends Singleton {
 
       /* assume bundled extensions are compatibile */
       if (!isBundled && manifest.engines?.lens) {
-        const appSemVerLatestImplied = appSemVer;
-
-        /* remove any prerelease tag so the extension's compatibility is not limited by it */
-        appSemVerLatestImplied.prerelease = [];
-        isCompatible = semver.satisfies(appSemVerLatestImplied, manifest.engines.lens);
+        /* include Lens's prerelease tag in the matching so the extension's compatibility is not limited by it */
+        isCompatible = semver.satisfies(appSemVer, manifest.engines.lens, { includePrerelease: true });
```

Comment:
Are you sure that that we want to set `includePrerelease: true` here?

Reply:
I think so, without it an extension will be deemed incompatible if the Lens version has a tag (and the extension's range does not have a tag, which would normally be the case). With it, Lens versions having a tag will be included in the matching criteria to the extensions range.

Do you see a problem?

Reply:
nope, maybe this should be factored out into a separate function that we have unit tests for. Maybe something with the signature:

```js
function checkCompatability(manifest: LensExtensionManifest, isBundled: boolean): boolean;
```

I guess it does make sense (a bit) that `includePrerelease: true` for the range `>= 5.0.1` would include `5.0.2-beta.1` but I think tests would still help.